// chalkee@1.4.0/dist/core/Chalkee.mjs
import { createStyleState as _, styleStateToAnsi as T, createReset as h, applyStyle as y } from "./styler.mjs";
import { attachSpecialBgMethods as w } from "./utils.mjs";
import { ChalkeeBase as p } from "./ChalkeeBase.mjs";
class i extends p {
  constructor(n, u = "") {
    if (super(), this._state = n || _(), u) {
      const e = T(this._state), s = h();
      this._accumulatedText = e + u + s;
    } else
      this._accumulatedText = "";
    const t = function(...e) {
      if (e.length === 1 && typeof e[0] == "string") {
        const s = e[0], c = new i(t._state, ""), a = d(s, t._state);
        return c._accumulatedText = t._accumulatedText + a, f(c, t), c;
      } else {
        if (e.length === 0)
          return t;
        if (e.length >= 1 && Array.isArray(e[0]) && typeof e[0][0] == "string" && "raw" in e[0]) {
          const s = e[0], c = e.slice(1);
          let a = s[0];
          for (let m = 0; m < c.length; m++)
            a += String(c[m]) + (s[m + 1] || "");
          const r = new i(t._state, ""), x = d(a, t._state);
          return r._accumulatedText = t._accumulatedText + x, f(r, t), r;
        } else {
          const s = e.map((r) => String(r)).join(""), c = new i(t._state, ""), a = d(s, t._state);
          return c._accumulatedText = t._accumulatedText + a, f(c, t), c;
        }
      }
    };
    Object.setPrototypeOf(t, i.prototype);
    const l = Symbol.for("nodejs.util.inspect.custom");
    return Object.defineProperty(t, l, {
      value: function(e, s, c) {
        return p.noColor ? this._accumulatedText.replace(/\x1b\[[0-9;]*m/g, "") : this._accumulatedText;
      },
      writable: !0,
      enumerable: !1,
      configurable: !0
    }), t._state = this._state, t._accumulatedText = this._accumulatedText, t;
  }
  call(...n) {
    if (n.length === 1 && typeof n[0] == "string") {
      const u = n[0], t = new i(this._state, ""), l = d(u, this._state);
      return t._accumulatedText = this._accumulatedText + l, t;
    } else {
      if (n.length === 0)
        return this;
      {
        const u = n.map((e) => String(e)).join(""), t = new i(this._state, ""), l = d(u, this._state);
        return t._accumulatedText = this._accumulatedText + l, t;
      }
    }
  }
}
function f(o, n) {
  n._isInBgMode && (o._isInBgMode = !0, w(o));
}
function d(o, n) {
  return y(o, n);
}
function b(o, n = "") {
  return new i(o, n);
}
function j(o, n) {
  return b(n, o);
}
export {
  i as Chalkee,
  j as createStyledFunction
};
