// chalkee@1.4.0/dist/core/styler.mjs
import { ChalkeeBase as s } from "./ChalkeeBase.mjs";
let c;
function t(r) {
  c = r;
}
const l = /* @__PURE__ */ new Map();
function d(r) {
  if (r.hexColorCode)
    return r.hexColorCode;
  if (r.rgbColorCode)
    return r.rgbColorCode;
  if (r.bgHexColorCode)
    return r.bgHexColorCode;
  if (r.bgRgbColorCode)
    return r.bgRgbColorCode;
  const o = [];
  for (const e of r.modifiers) {
    const C = c ? c(e) : void 0;
    C !== void 0 && o.push(C);
  }
  for (const e of r.colors)
    o.push(e.ansiCode);
  for (const e of r.backgroundColors)
    o.push(e.ansiCode);
  if (o.length === 0)
    return "";
  const i = o.join(";");
  if (l.has(i))
    return l.get(i);
  const n = `\x1B[${i}m`;
  return l.set(i, n), n;
}
function g() {
  return "\x1B[0m";
}
function u(r, o) {
  if (s.noColor)
    return r;
  const i = d(o), n = g();
  return o.autoSpacing && o.previousStyles && (o.previousStyles.colors.length > 0 || o.previousStyles.modifiers.length > 0 || o.previousStyles.backgroundColors.length > 0) ? " " + d(o) + r + n : i + r + n;
}
function p(r, o) {
  let i = r.modifiers;
  if (o.modifiers.length > 0) {
    const e = o.modifiers.length > 0 ? [...r.modifiers, ...o.modifiers] : r.modifiers;
    i = e.length > 0 && new Set(e).size < e.length ? [...new Set(e)] : e;
  }
  const n = {
    colors: r.colors.length > 0 || o.colors.length > 0 ? [...r.colors, ...o.colors] : r.colors,
    modifiers: i,
    backgroundColors: r.backgroundColors.length > 0 || o.backgroundColors.length > 0 ? [...r.backgroundColors, ...o.backgroundColors] : r.backgroundColors,
    isOpen: o.isOpen,
    previousStyles: r,
    autoSpacing: o.autoSpacing
  };
  return o.hexColorCode && (n.hexColorCode = o.hexColorCode), o.rgbColorCode && (n.rgbColorCode = o.rgbColorCode), o.bgHexColorCode && (n.bgHexColorCode = o.bgHexColorCode), o.bgRgbColorCode && (n.bgRgbColorCode = o.bgRgbColorCode), n;
}
function h() {
  return {
    colors: [],
    modifiers: [],
    backgroundColors: [],
    isOpen: !1,
    autoSpacing: !1
  };
}
export {
  u as applyStyle,
  g as createReset,
  h as createStyleState,
  p as mergeStyleStates,
  t as setModifierLookup,
  d as styleStateToAnsi
};
