// chalkee/dist/index.mjs
import { createStyleState as e } from "./core/styler.mjs";
import { createStyledFunction as r } from "./core/Chalkee.mjs";
import { Chalkee as l } from "./core/Chalkee.mjs";
import { parseHex as n, rgbToAnsi256 as B } from "./core/registry.mjs";
import "./core/callable-helpers.mjs";
import { bgBlack as d, bgBlackBright as m, bgBlue as y, bgBlueBright as p, bgCyan as s, bgCyanBright as u, bgGreen as x, bgGreenBright as f, bgMagenta as k, bgMagentaBright as w, bgRed as C, bgRedBright as R, bgWhite as S, bgWhiteBright as G, bgYellow as H, bgYellowBright as M, black as W, blackBright as Y, blue as v, blueBright as A, cyan as F, cyanBright as T, gray as j, green as q, greenBright as z, grey as D, magenta as E, magentaBright as I, red as J, redBright as K, white as L, whiteBright as N, yellow as O, yellowBright as P } from "./core/ansi-colors.mjs";
import { as as U, b as V, bg as X, bgHex as Z, bgRgb as _, bold as $, d as ee, dim as re, hex as ge, hidden as te, i as ie, inverse as be, italic as ae, r as he, reset as le, rgb as oe, s as ne, strikethrough as Be, u as ce, underline as de } from "./core/modifiers.mjs";
const b = r("", e());
export {
  l as Chalkee,
  U as as,
  V as b,
  X as bg,
  d as bgBlack,
  m as bgBlackBright,
  y as bgBlue,
  p as bgBlueBright,
  s as bgCyan,
  u as bgCyanBright,
  x as bgGreen,
  f as bgGreenBright,
  Z as bgHex,
  k as bgMagenta,
  w as bgMagentaBright,
  C as bgRed,
  R as bgRedBright,
  _ as bgRgb,
  S as bgWhite,
  G as bgWhiteBright,
  H as bgYellow,
  M as bgYellowBright,
  W as black,
  Y as blackBright,
  v as blue,
  A as blueBright,
  $ as bold,
  F as cyan,
  T as cyanBright,
  ee as d,
  b as default,
  re as dim,
  j as gray,
  q as green,
  z as greenBright,
  D as grey,
  ge as hex,
  te as hidden,
  ie as i,
  be as inverse,
  ae as italic,
  E as magenta,
  I as magentaBright,
  n as parseHex,
  he as r,
  J as red,
  K as redBright,
  le as reset,
  oe as rgb,
  B as rgbToAnsi256,
  ne as s,
  Be as strikethrough,
  ce as u,
  de as underline,
  L as white,
  N as whiteBright,
  O as yellow,
  P as yellowBright
};
