// chalkee/dist/core/ansi-colors.mjs
import { ChalkeeBase as c } from "./ChalkeeBase.mjs";
import { mergeStyleStates as a, createStyleState as l } from "./styler.mjs";
import { createStyledFunction as e } from "./Chalkee.mjs";
import { getColor as h } from "./registry.mjs";
const i = (r, o = !1) => {
  Object.defineProperty(c.prototype, r, {
    get: function() {
      return o ? this.createBgColor(r) : this.createColor(r);
    },
    enumerable: !0,
    configurable: !0
  });
};
function b() {
  const r = [
    "red",
    "green",
    "blue",
    "yellow",
    "magenta",
    "cyan",
    "white",
    "black",
    "gray",
    "grey",
    "redBright",
    "greenBright",
    "blueBright",
    "yellowBright",
    "magentaBright",
    "cyanBright",
    "whiteBright",
    "blackBright"
  ], o = [
    "bgRed",
    "bgGreen",
    "bgBlue",
    "bgYellow",
    "bgMagenta",
    "bgCyan",
    "bgWhite",
    "bgBlack",
    "bgRedBright",
    "bgGreenBright",
    "bgBlueBright",
    "bgYellowBright",
    "bgMagentaBright",
    "bgCyanBright",
    "bgWhiteBright",
    "bgBlackBright"
  ];
  for (const n of r)
    i(n, !1);
  for (const n of o)
    i(n, !0);
}
b();
const t = l();
function g(r) {
  const o = h(r);
  if (!o)
    return e("", t);
  const n = a(t, {
    ...t,
    colors: [o],
    isOpen: !0,
    autoSpacing: !1
  });
  return e("", n);
}
const f = g("red"), w = g("green"), d = g("blue"), m = g("yellow"), C = g("magenta"), k = g("cyan"), p = g("white"), S = g("black"), M = g("gray"), G = g("grey"), R = g("redBright"), W = g("greenBright"), Y = g("blueBright"), F = g("yellowBright"), O = g("magentaBright"), P = g("cyanBright"), j = g("whiteBright"), x = g("blackBright"), A = e("", t).bgRed, D = e("", t).bgGreen, q = e("", t).bgBlue, v = e("", t).bgYellow, z = e("", t).bgMagenta, E = e("", t).bgCyan, H = e("", t).bgWhite, I = e("", t).bgBlack, J = e("", t).bgRedBright, K = e("", t).bgGreenBright, L = e("", t).bgBlueBright, N = e("", t).bgYellowBright, Q = e("", t).bgMagentaBright, T = e("", t).bgCyanBright, U = e("", t).bgWhiteBright, V = e("", t).bgBlackBright;
export {
  I as bgBlack,
  V as bgBlackBright,
  q as bgBlue,
  L as bgBlueBright,
  E as bgCyan,
  T as bgCyanBright,
  D as bgGreen,
  K as bgGreenBright,
  z as bgMagenta,
  Q as bgMagentaBright,
  A as bgRed,
  J as bgRedBright,
  H as bgWhite,
  U as bgWhiteBright,
  v as bgYellow,
  N as bgYellowBright,
  S as black,
  x as blackBright,
  d as blue,
  Y as blueBright,
  k as cyan,
  P as cyanBright,
  M as gray,
  w as green,
  W as greenBright,
  G as grey,
  C as magenta,
  O as magentaBright,
  f as red,
  R as redBright,
  p as white,
  j as whiteBright,
  m as yellow,
  F as yellowBright
};
