// chalkee/dist/core/modifiers.mjs
import { ChalkeeBase as i } from "./ChalkeeBase.mjs";
import { setModifierLookup as d, createStyleState as a } from "./styler.mjs";
import { createStyledFunction as e } from "./Chalkee.mjs";
const b = {
  reset: 0,
  bold: 1,
  dim: 2,
  italic: 3,
  underline: 4,
  inverse: 7,
  hidden: 8,
  strikethrough: 9
};
function g(r) {
  return b[r];
}
d(g);
const l = (r, o) => {
  Object.defineProperty(i.prototype, r, {
    get: function() {
      return this[o](r);
    },
    enumerable: !0,
    configurable: !0
  });
}, n = (r, o) => {
  Object.defineProperty(i.prototype, r, {
    get: function() {
      return this[o];
    },
    enumerable: !0,
    configurable: !0
  });
}, u = (r, o) => {
  Object.defineProperty(i.prototype, r, {
    get: function() {
      return this[o]();
    },
    enumerable: !0,
    configurable: !0
  });
}, c = (r, o) => {
  Object.defineProperty(i.prototype, r, {
    get: function() {
      return (...s) => this[o](...s);
    },
    enumerable: !0,
    configurable: !0
  });
};
function f(r) {
  const o = ["bold", "dim", "italic", "underline", "strikethrough", "inverse", "hidden"];
  for (const s of o)
    l(s, "createModifier");
  n("b", "bold"), n("d", "dim"), n("i", "italic"), n("u", "underline"), n("s", "strikethrough"), u("reset", "createReset"), n("r", "reset"), u("as", "createAutoSpacing"), Object.defineProperty(i.prototype, "bg", {
    get: function() {
      return this.createBgMode();
    },
    enumerable: !0,
    configurable: !0
  }), c("hex", "createHexColor"), c("rgb", "createRgbColor"), c("bgHex", "createBgHexColor"), c("bgRgb", "createBgRgbColor");
}
f();
const t = a(), y = e("", t).bold, x = e("", t).dim, P = e("", t).italic, k = e("", t).underline, M = e("", t).strikethrough, S = e("", t).inverse, C = e("", t).hidden, R = e("", t).reset, O = e("", t).b, j = e("", t).d, B = e("", t).i, H = e("", t).u, I = e("", t).s, v = e("", t).r, A = e("", t).as, F = e("", t).bg, D = e("", t).hex, E = e("", t).rgb, L = e("", t).bgHex, N = e("", t).bgRgb;
export {
  b as ANSI_MODIFIERS,
  A as as,
  O as b,
  F as bg,
  L as bgHex,
  N as bgRgb,
  y as bold,
  j as d,
  x as dim,
  g as getModifier,
  D as hex,
  C as hidden,
  B as i,
  S as inverse,
  P as italic,
  v as r,
  R as reset,
  E as rgb,
  I as s,
  M as strikethrough,
  H as u,
  k as underline
};
